pandas/core/groupby/base.py

```diff
@@ -98,6 +98,8 @@ def _gotitem(self, key, ndim, subset=None):
     [
         "all",
         "any",
+        # `corrwith` is a reduction as long as `other` is a Series
```

Comment:
Hmm maybe not thinking clearly but when is this not a reduction? I don’t quite understand the comment

Reply:
When `other` is a `DataFrame` and it has different column names, then it is not strictly a reduction, as it produces more columns than `self` originally has.

Example:
```python
>>> ser = pd.Series([1,2,3,1,2,3], name='test')
>>> df1 = pd.DataFrame(dict(key=list('aaabbb'), value1=[1,2,3,3,2,1], value2=[1,3,2,2,3,1]))
>>> df2 = pd.DataFrame(dict(col1=[1,2,3,3,2,1], col2=[1,3,2,2,3,1]))
>>> df1
  key  value1  value2
0   a       1       1
1   a       2       3
2   a       3       2
3   b       3       2
4   b       2       3
5   b       1       1
>>> df2
   col1  col2
0     1     1
1     2     3
2     3     2
3     3     2
4     2     3
5     1     1

>>> ser
0    1
1    2
2    3
3    1
4    2
5    3
Name: test, dtype: int64
```

```python
# groupby.corrwith is a reduction when `other` is a Series (the name "test" didn't matter)
>>> df1.groupby('key').corrwith(ser)
     value1  value2
key
a       1.0     0.5
b      -1.0    -0.5

# groupby.corrwith is a reduction when `other` is a DataFrame and has different column names
>>> df1.groupby('key').corrwith(df2)
     col1  col2  value1  value2
key
a     NaN   NaN     NaN     NaN
b     NaN   NaN     NaN     NaN

# not a reduction even if `other` only has 1 column (and that column has a different name),
>>> df1[['key', 'value1']].groupby('key').corrwith(df2[['col1']])
     col1  value1
key
a     NaN     NaN
b     NaN     NaN

# when they have the same column names, it is a reduction
>>> df1.groupby('key').corrwith(df1)
     value1  value2
key
a       1.0     1.0
b       1.0     1.0
```

Initially, I thought I'd put something like this 
```
`corrwith` is a reduction as long as `other` is a `Series` or is a `DataFrame` with the same numerical columns
```
However, the above is a bit confusing and not entirely precise. In fact, it is pretty hard to precisely describe when `corrwith` is a reduction without confusion. See below
```python
# the order of columns, and non-numeric columns ("key") didn't matter
>>> df1.groupby('key').corrwith(df1[['value2', 'value1']])
     value1  value2
key
a       1.0     1.0
b       1.0     1.0

# still a reduction even though `other` has less numerical columns
>>> df1.groupby('key').corrwith(df1[['value2']])
     value2  value1
key
a       1.0     NaN
b       1.0     NaN

# then there is the complication of drop=True
>>> df1.groupby('key').corrwith(df1[['value2']], drop=True)
     value2
key
a       1.0
b       1.0
```

Perhaps we can think of something like this?
```
`corrwith` is a reduction as long as `other` is a `Series` or is a `DataFrame` whose numerical columns are a subset of the numerical columns of `self`
```

Reply:
Hmm OK. I get what you are saying but I don't think the comment really adds value or expresses that. Can you remove?

Reply:
ok. done